packages/scripts/src/clusterize.ts

```diff
@@ -0,0 +1,89 @@
+import dbscan from "@cdxoo/dbscan";
+import { strict as assert } from "assert";
+
+/**
+  Based on https://www.npmjs.com/package/@cdxoo/dbscan
+ */
+export type DbscanOptions = {
+  /**
+    Maximum distance between datapoints. Determine if a datapoint is in a
+    cluster or not. Default is 1.0.
+   */
+  epsilon: number;
+
+  /**
+    Custom function to compare calculated distance and epsilon. Default is
+    (dist, e) => (dist < e)
+   */
+  epsilonCompare: (distance: number, epsilon: number) => boolean;
```

Comment:
What's the use case for a custom epsilon comparison? Isn't a core point neighborhood defined as d < e by the algorithm?

Reply:
This just comes from the dbscan library and I suspect it's so that you can do stuff like compare an 'actual length' epsilon with a 'magnitude' (e.g. `e < sqrt(distance)`). Note I don't bother sqrting the distances and just expect e to fit accordingly. Maybe you can also do `d > e` to make anti-clusters. I dunno! I didn't see a reason NOT to expose it so I just kept it in.